### PR TITLE
RenderToNodeStream

### PR DESCRIPTION
**Pull request by strues**
This adds ReactDOM.RenderToNodeStream for server rendering (issue #524)

### CODE REVIEW

**Comment by williamoliveira** [comment]
Instead of rendering `app` to string a string, you could pass the component to `ServerHTML` and than to `HTML` and insert it there so everything is stream rendered

on `HTML`:
```
- <div id="app" dangerouslySetInnerHTML={{ __html: appBodyString }} />
+ <div id="app">{reactApp}</div>
```

File: server/middleware/reactApplication/index.js [hunk 1]
```diff
@@ -1,36 +1,43 @@
-import React from 'react';
-import Helmet from 'react-helmet';
-import { renderToString, renderToStaticMarkup } from 'react-dom/server';
-import { StaticRouter } from 'react-router-dom';
-import { AsyncComponentProvider, createAsyncContext } from 'react-async-component';
-import asyncBootstrapper from 'react-async-bootstrapper';
+import React from "react";
```
Comment (SleepWalker) [inline server/middleware/reactApplication/index.js:1]: Why do you use double quotes? Have you linted your code?

Comment (ctrlplusb) [reply server/middleware/reactApplication/index.js:1]: Looks like `prettier` didn't run properly?  FYI all I have updated prettier.  We no longer use `prettier-eslint`.  Just use native `prettier` and `eslint` separately.  I installed the `eslint-config-prettier` plugin which disables all the style rules so we can still use the airbnb config.

**Comment by ctrlplusb** [comment]
Thanks for picking this up @strues - would you mind rebasing this PR or creating a new one and cherry picking the commits from this one?  I'd like us to start running PRs using a rebase strategy to avoid noise.

**Comment by strues** [comment]
I'll recreate and rebase top of the most recent `next` branch. I've been cherry picking and dealing with conflicts too long that its not worth it :)

**Pull request closed without merging**